### PR TITLE
remove /projects/

### LINKED ISSUE #6221: Remove /projects/
All related code should be removed, that page isn't useful https://readthedocs.org/projects/

https://github.com/readthedocs/readthedocs.org/blob/1ce31662650d6defa434587ec0325f044052ee72/readthedocs/urls.py#L56-L58

### PR DESCRIPTION
**Pull request by iambenzo**
Fixes #6221

I am relatively new to development with Django but to remove the publicly accessible `/projects` path, only a small amount of removals seemed to be required.

I did not remove the [ProjectIndex](https://github.com/readthedocs/readthedocs.org/blob/1ce31662650d6defa434587ec0325f044052ee72/readthedocs/projects/views/public.py#L40 ) class, as it seems that the tags functionality is dependant on it. If you're happy for the tags to go, I will be happy to update my PR.

I did not remove the [project_urls](https://github.com/readthedocs/readthedocs.org/blob/1ce31662650d6defa434587ec0325f044052ee72/readthedocs/urls.py#L56-L58) section, as clicking on a project as a logged in user takes you to a `/projects/<project_name>` route.

Please let me know how I did in meeting your expectations and let me know if I can do any better.

### CODE REVIEW

**Comment by stsewd** [comment]
Thanks for the pull request!

> I did not remove the ProjectIndex class, as it seems that the tags functionality is dependant on it. If you're happy for the tags to go, I will be happy to update my PR.

I'd say that section can go too.

Looks like it links from https://readthedocs.org/projects/docs/ (the tags section) https://readthedocs.org/projects/docs/

Maybe @davidfischer has another opinion here

But if the section stays, you shouldn't remove the template, as that view still uses it.

**Comment by stsewd** [comment]
Sorry, wrong button

**Comment by iambenzo** [comment]
Okay, thanks - I'll wait for input from @davidfischer before I make any further updates.

**Comment by davidfischer** [comment]
> Maybe @davidfischer has another opinion here

At least right now I don't think we should remove browsing projects by tag (eg. https://readthedocs.org/projects/tags/python/). Compared with listing all projects, at least tags have a use which is finding similar projects.

I also wouldn't mind if `/projects/` redirected to `/dashboard/` but that can come in a separate PR.

**Comment by iambenzo** [comment]
Okay, thanks. In that case, I think my course of action is to reinstate the template I removed and all should be well.

**Comment by stsewd** [comment]
@iambenzo can you please merge master into this branch? Test on travis have been fixed

**Comment by iambenzo** [comment]
Sure thing.

**Comment by stsewd** [comment]
We can now remove this conditional, as we always expect a tag now, and probably rename the class https://github.com/readthedocs/readthedocs.org/blob/2e7678124fb9a3f976948e144030fc2cda0a556f/readthedocs/projects/views/public.py#L50-L50

**Comment by iambenzo** [comment]
Okie dokie - I'll get on that a little later

File: readthedocs/projects/views/public.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 mimetypes.add_type('application/epub+zip', '.epub')
 
 
-class ProjectIndex(ListView):
+class ProjectTagIndex(ListView):
```
Comment (davidfischer) [inline readthedocs/projects/views/public.py]: I'm just noting that this change will require a corresponding change to the private Read the Docs corporate repository.

File: readthedocs/projects/views/public.py [hunk 2]
```diff
@@ -47,11 +47,8 @@ def get_queryset(self):
...
 
-        if self.kwargs.get('tag'):
-            self.tag = get_object_or_404(Tag, slug=self.kwargs.get('tag'))
-            queryset = queryset.filter(tags__slug__in=[self.tag.slug])
-        else:
-            self.tag = None
+        self.tag = get_object_or_404(Tag, slug=self.kwargs.get('tag'))
+        queryset = queryset.filter(tags__slug__in=[self.tag.slug])
 
         if self.kwargs.get('username'):
```
Comment (davidfischer) [inline readthedocs/projects/views/public.py]: It doesn't seem like this is used anymore. As a separate change in a separate PR, this could probably be removed.

Comment (stsewd) [reply readthedocs/projects/views/public.py]: #6250
